lib/grape-swagger/endpoint.rb

```diff
@@ -282,14 +282,25 @@ def default_type(params)
       end
     end
 
-    def parse_request_params(params)
-      array_key = nil
+    def parse_request_params(params, settings)
+      array_keys = []
       params.select { |param| public_parameter?(param) }.each_with_object({}) do |param, memo|
         name, options = *param
         param_type = options[:type]
         param_type = param_type.to_s unless param_type.nil?
-        array_key = name.to_s if param_type_is_array?(param_type)
-        options[:is_array] = true if array_key && name.start_with?(array_key)
+
+        array_keys << name.to_s if param_type_is_array?(param_type)
+
+        keys = array_keys.find_all { |key| name.start_with? key }
+        if keys.any?
+          options[:is_array] = true
+          if settings[:array_use_braces] && !(options[:documentation] && options[:documentation][:param_type] == 'body')
+            keys.sort.reverse_each do |key|
```

Comment:
I'm doing sort and reverse each here so that we will start replacing with longer strings, like with `keys == ['a[b]', 'a']`: `a[b][c]` => `a[b][][c]` and only then => `a[][b][][c]`. Otherwise it will replace it to `a[][b][c]` and then wouldn't find `a[b]` to replace